debounce.go

```diff
@@ -0,0 +1,105 @@
+// Pacage debounce provides a number of debounce patterns as a set of easy to
+// use functions.
+package debounce
+
+import (
+	"sync"
+	"time"
+)
+
+// New returns a debounced function that delays invoking f until after wait time
+// has elapsed since the last time the debounced function was invoked.
+//
+// The returned cancel function can be used to cancel any pending invocation of
+// f.
+//
+// Both debounced and cancel functions are safe for concurrent use in
+// goroutines, and can both be called multiple times.
+//
+// The debounced function does not wait for f to complete, so f needs to be
+// thread-safe as it may be invoked again before the previous invocation
+// completes.
+func New(wait time.Duration, f func()) (debounced func(), cancel func()) {
```

Comment:
Consider making a note on whether cancel needs to be called or nor.
Since there isn't a go routine like the ctx cancelation I guess it's safe to ignore.